labs/FineTuning/HuggingFaceExample/01_finetuning/Finetune-TinyLlama-1.1B.ipynb

```diff
@@ -0,0 +1,1068 @@
+    "\n",
+    "In the `hyperparameters` section, you can see the specific command-line arguments that are used to control the behavior of the `finetune_llama.py` training script. Notably:\n",
+    "- *model_id*: specifies which model you will be fine-tuning, in this case a recent checkpoint from the TinyLlama-1.1B project\n",
+    "- *tokenizer_id*: specifies which tokenizer you will used to tokenize the dataset examples during training\n",
+    "- *output_dir*: directory in which the fine-tuned model will be saved. Here we use the SageMaker-specific `/opt/ml/model` directory. At the end of the training job, SageMaker automatically copies the contents of this directory to the output S3 bucket\n",
+    "- *tensor_parallel_size*: the tensor parallel degree for which we want to use for training. In this case we use '2' to shard the model across the 2 NeuronCores available in the trn1.2xlarge instance\n",
+    "- *bf16*: request BFloat16 training\n",
+    "- *per_device_train_batch_size*: the microbatch size to be used for fine-tuning\n",
+    "- *gradient_accumulation_steps*: how many steps for which gradients will be accumulated between updates\n",
+    "- *max_steps*: the maximum number of steps of fine-tuning that we want to perform\n",
+    "- *lora_r*, *lora_alpha*, *lora_dropout*: the LoRA rank, alpha, and dropout values to use during fine-tuning\n",
+    "\n",
+    "The below estimator has been pre-configured for you, so you do not need to make any changes."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "id": "9e28014c-4d0b-452b-9bde-44aa10e61bb6",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "# Set up the PyTorch estimator\n",
+    "# Note that the hyperparameters are just command-line args passed to the finetune_llama.py script to control its behavior\n",
+    "\n",
+    "pt_estimator = PyTorch(\n",
+    "        entry_point=\"finetune_llama.py\",\n",
+    "        source_dir=\"./assets\",\n",
+    "        role=get_execution_role(),\n",
+    "        instance_count=1,\n",
+    "        instance_type=\"ml.trn1.2xlarge\",\n",
+    "        disable_profiler=True,\n",
+    "        output_path=f\"s3://{default_bucket}/neuron_events2025\",\n",
+    "        base_job_name=\"trn1-tinyllama\",\n",
+    "        sagemaker_session=sess,\n",
+    "        code_bucket=f\"s3://{default_bucket}/neuron_events2025_code\",\n",
+    "        checkpoint_s3_uri=f\"s3://{default_bucket}/neuron_events_output\",\n",
+    "        image_uri=training_image,\n",
+    "        distribution={\"torch_distributed\": {\"enabled\": True}},\n",
+    "        environment={\"FI_EFA_FORK_SAFE\": \"1\", \"WANDB_DISABLED\": \"true\"},\n",
+    "        disable_output_compression=True,\n",
+    "        hyperparameters={\n",
+    "            \"model_id\": \"TinyLlama/TinyLlama-1.1B-Chat-v1.0\",\n",
```

Comment:
Does the training job script download the model to the ephemeral instance? Might be helpful to call out.